Scripts/Runtime/Util/Systems/PlayerLoopUtil.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Reflection;
+using Unity.Entities;
+using UnityEngine.LowLevel;
+
+namespace Anvil.Unity.DOTS.Util
+{
+    /// <summary>
+    /// Helper class for exposing and dealing with some <see cref="PlayerLoopSystem"/> internals.
+    /// </summary>
+    public static class PlayerLoopUtil
+    {
+        private static readonly Type COMPONENT_SYSTEM_GROUP_TYPE = typeof(ComponentSystemGroup);
+        private static readonly MethodInfo s_ScriptBehaviourUpdateOrder_IsDelegateForWorldSystem_MethodInfo = typeof(ScriptBehaviourUpdateOrder).GetMethod("IsDelegateForWorldSystem", BindingFlags.Static | BindingFlags.NonPublic);
+        private static readonly IsDelegateForWorldSystemDelegate s_IsDelegateForWorldSystem = (IsDelegateForWorldSystemDelegate)Delegate.CreateDelegate(typeof(IsDelegateForWorldSystemDelegate), s_ScriptBehaviourUpdateOrder_IsDelegateForWorldSystem_MethodInfo);
+        private static PropertyInfo s_DummyDelegateWrapper_System;
+        
+        private delegate bool IsDelegateForWorldSystemDelegate(World world, ref PlayerLoopSystem playerLoopSystem);
+        
+        /// <summary>
+        /// Checks if a <see cref="PlayerLoopSystem"/> is part of a given <see cref="World"/>
+        /// </summary>
+        /// <param name="playerLoopSystem">The <see cref="PlayerLoopSystem"/> to check.</param>
+        /// <param name="world">The <see cref="World"/> that may contain the <see cref="PlayerLoopSystem"/></param>
+        /// <returns>
+        /// True if part of the world.
+        /// False if not part of the world.
+        /// False if the <see cref="PlayerLoopSystem"/> is a phase instead of a <see cref="ComponentSystemGroup"/>
+        /// </returns>
+        public static bool IsPlayerLoopSystemPartOfWorld(ref PlayerLoopSystem playerLoopSystem, World world)
+        {
+            return playerLoopSystem.updateDelegate?.Target != null && s_IsDelegateForWorldSystem(world, ref playerLoopSystem);
+        }
+
+        /// <summary>
+        /// Given a <see cref="PlayerLoopSystem"/>, gets the <see cref="ComponentSystemGroup"/> associated.
+        /// This is a "safe" method, it does not assume there is a <see cref="ComponentSystemGroup"/> to get
+        /// and may return null.
```

Comment:
Thoughts on using the try get pattern for the methods like this in this class?

Reply:
Yeah ok, changed.